### PR TITLE
[#IOCIT-238] migration to eslint

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* added config files for eslint
* removed tslint config files and references to it
* added lint script
* edited codebase to solve linting problems
* added the necessary dependencies for eslint
* removed dependencies of tslint

#### Motivation and Context

all projects needs to be migrated to eslint

#### How Has This Been Tested?

lint, build and tests passing

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: GetServiceByRevision/handler.ts [hunk 1]
```diff
@@ -121,29 +123,30 @@ const getServiceByRevisionTask = (
...
-      TE.mapLeft(error =>
-        ResponseErrorQuery("Error while retrieving the service", error)
-      ),
-      TE.toUnion
-    )();
-}
+): IGetServiceByRevisionHandler => async (
+  serviceId,
+  version
+): Promise<IGetServiceByRevisionHandlerRet> =>
```
Comment (BurnedMarshal) [inline GetServiceByRevision/handler.ts:131]:
```suggestion
): ReturnType<IGetServiceByRevisionHandler> =>
```

File: GetServiceMetadata/handler.ts [hunk 2]
```diff
@@ -82,44 +87,42 @@ const findService = (serviceModel: ServiceModel, serviceId: NonEmptyString) =>
...
-                  "The service you requested doesn't have metadata attribute."
-                )
-        )
-      ),
-      TE.toUnion
-    )();
-}
+): IGetServiceMetadataHandler => async (
+  serviceId
+): Promise<IGetServiceMetadataHandlerRet> =>
```
Comment (BurnedMarshal) [inline GetServiceMetadata/handler.ts:94]:
```suggestion
): ReturnType<IGetServiceMetadataHandler> =>
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM. Just a consideration on return type for handlers. Maybe is better use `ReturnType` when an interface for the handler is defined to avoid to copy past types on changes on multiple lines.

**Comment by arcogabbo** [comment]
> LGTM. Just a consideration on return type for handlers. Maybe is better use `ReturnType` when an interface for the handler is defined to avoid to copy past types on changes on multiple lines.

nice catch. editing now 👀

**Pull request merged**